portal-api/api-tests/v1/users/put-update-user.api-test.js

```diff
@@ -164,7 +164,7 @@ describe('a user', () => {
 
         await as(createdUser).put(updatedUserCredentials).to(`/v1/users/${createdUser._id}`);
 
-        const { status } = await as(aNonAdmin).get(`/v1/users/${createdUser._id}`);
+        const { status } = await as(createdUser).get(`/v1/users/${createdUser._id}`);
 
         expect(status).toEqual(200);
```

Comment:
This test actually never checked what it was supposed to in the first place. It makes a request to `PUT users/:id` to update the user's password but then checks the status of a request to `GET users/:id`, which will be `200` regardless of the outcome of the request to update the password. The password update was actually failing with `401` because the `createdUser` was not being set up with a bearer token correctly. This test should instead check that the update password request succeeds with a `200`, that the user can't log in with their old password and that they can log in with their new password.

Reply:
The test should look something like this:
`it('a user can update their own password', async () => {
        const updatedPassword = 'updatedPassword1!';
        const updatedUserCredentials = {
          currentPassword: initialPassword,
          password: updatedPassword,
          passwordConfirm: updatedPassword,
        };
        const { status: updatePasswordStatus } = await as(createdUser).put(updatedUserCredentials).to(`/v1/users/${createdUser._id}`);
        expect(updatePasswordStatus).toEqual(200);
        const { status: loginWithInitalPasswordStatus } = await as().post({ username: createdUser.username, password: initialPassword }).to('/v1/login');
        expect(loginWithInitalPasswordStatus).toEqual(401);
        const { status: loginWithUpdatedPasswordStatus } = await as().post({ username: createdUser.username, password: updatedPassword }).to('/v1/login');
        expect(loginWithUpdatedPasswordStatus).toEqual(200);
});`

Reply:
Similarly, the `as admin` => `a user's password can be updated` test should be changed to follow this pattern.

Reply:
In order to make this work, you'll need to set up the bearer token correctly for the `createdUser` by adding the following lines at the end of the `beforeEach` block:

```
const initialUserCredentials = {
      password: initialPassword,
      passwordConfirm: initialPassword,
    };
await as(anAdmin).put(initialUserCredentials).to(`/v1/users/${createdUser._id}`);
const { token } = await createLoggedInUserSession(createdUser);
createdUser.token = token;
```

Reply:
I had added `initialPassword` and imported `createLoggedInUserSession`